modules/ballerina-core/src/main/java/org/ballerinalang/util/codegen/ResourceInfo.java

```diff
@@ -31,6 +31,11 @@
 
     private ServiceInfo serviceInfo;
 
+    // below are cache values.
+    private String[] httpMethods;
```

Comment:
This is wrong IMO. These cache values are only for the HTTP resources, therefore cannot be stored in the ResourceInfo class.